### PR TITLE
Add Comment to DatabaseColumn and DatabaseTable

### PR DESCRIPTION
**Pull request by ErikEJ**
and Scaffold comment from SQL Server MS_Description

Part of #16305

### CODE REVIEW

File: src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs [hunk 1]
```diff
@@ -551,7 +613,8 @@ private static string EscapeLiteral(string s)
             IReadOnlyList<DatabaseTable> tables,
             string tableFilter,
             string viewFilter,
-            IReadOnlyDictionary<string, (string storeType, string typeName)> typeAliases)
+            IReadOnlyDictionary<string, (string storeType, string typeName)> typeAliases,
+            IReadOnlyDictionary<string, string> comments)
```
Comment (bricelam) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: It feels awkward to pass these in. Why not just add a JOIN to the query below?

**Comment by ErikEJ** [comment]
@bricelam Question: For scaffolding with DataAnnotations, is the plan to use [Description] attribute?

**Comment by bricelam** [comment]
I don't think we have a corresponding annotation yet; we should file an issue. So the plan for now would be to scaffold `HasComment` always (whether -DataAnnotations is specified or not)

**Comment by ErikEJ** [comment]
@bricelam I have updated RelationalScaffoldingModelFactory - but unsure if those changes are enough - and fail to find a relevant test.

**Comment by bricelam** [comment]
Thanks! I'll look into testing

**Comment by bricelam** [comment]
(Added some tests, made C# generation better, and rebased)

**Comment by bricelam** [comment]
Ah, just realized this is for entity type too 😄 Want to add tests for those and fix that C#?

**Comment by ErikEJ** [comment]
@bricelam I have tried 🤡

**Comment by ErikEJ** [comment]
But tests are failing..

File: test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs [hunk 2]
```diff
@@ -159,13 +159,32 @@ public void Comments_use_fluent_api()
                     }),
                 new ModelCodeGenerationOptions(),
                 code => Assert.Contains(
-                    ".HasColumn(\"An int property\")",
```
Comment (bricelam) [inline test/EFCore.Design.Tests/Scaffolding/Internal/CSharpDbContextGeneratorTest.cs]: 🤦‍♂ My bad.

**Comment by bricelam** [comment]
Updated. We needed to pass the fully-qualified name to `FindEntityType`

**Comment by ErikEJ** [comment]
Great!!!

File: src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs [hunk 3]
```diff
@@ -388,6 +389,13 @@ private void GenerateEntityType(IEntityType entityType, bool useDataAnnotations)
 
             lines.AddRange(GenerateAnnotations(annotations.Except(annotationsToRemove)));
 
+            if (entityType.GetComment() != null)
+            {
+                lines.Add(
+                    $".{nameof(RelationalPropertyBuilderExtensions.HasComment)}" +
```
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: Incorrect method reference

**Comment by ErikEJ** [comment]
Updated - test failures seem unrelated

**Pull request merged**